katz/src/main/kotlin/katz/Id.kt

```diff
@@ -0,0 +1,31 @@
+package katz
+
+class Id<out A>(val value: A): HK<Id.F, A> {
+
+    class F
+
+    inline fun <B> map(f: (A) -> B): Id<B> = Id(f(value))
+
+    inline fun <B> flatMap(f: (A) -> Id<B>): Id<B> = f(value)
+
+    companion object : IdInstances
+}
+
+/**
+ * Companion interface for [Id] type
+ * outside due to https://youtrack.jetbrains.com/issue/KT-10532
+ */
+private interface IdInstances {
+    fun <A> HK<Id.F, A>.ev(): Id<A> = this as Id<A>
+
+    fun monad() = object : Monad<Id.F> {
```

Comment:
How many objects does this create? One for the whole application or one per call? If it is one per call it will be probably better to extract it to a top level object.